src/components/userProfile/CareerRow.js

```diff
@@ -10,73 +11,129 @@ class CareerRow extends Component {
+      schoolName: undefined,
+      major: undefined,
+      graduationYear: undefined,
+      companyName: undefined,
+      position: undefined,
+      period: undefined,
       loaded: false,
     };
   }
 
-  componentWillMount() {
-    let schoolName = this.state.schoolName;
-    let major = this.state.major;
+  componentDidMount() {
+    if (this.props.sectionID === 'Experience') {
+      this.setState({
+        companyName: this.getCompanyName(),
+        position: this.getPosition(),
+        period: this.getPeriod(),
+        loaded: true,
+      });
+    } else if (this.props.sectionID === 'Education') {
+      this.setState({
+        schoolName: this.getSchoolName(),
+        major: this.getMajor(),
+        graduationYear: this.graduationYear(),
+        loaded: true,
+      });
+    }
+  }
+
+  getCompanyName() {
     let companyName = this.state.companyName;
-    let position = this.state.period;
-    let identifier = this.state.identifier;
-    let period = this.state.period;
 
     if (this.props.dataSource.employer) {
       companyName = this.props.dataSource.employer.name;
-      identifier = 1;
-      if (this.props.dataSource.position) {
-        position = '| ' + this.props.dataSource.position.name;
-      }
+      return companyName;
+    }
 
-      if (this.props.dataSource.start_date) {
-        period = this.props.dataSource.start_date + ' - ' + this.props.dataSource.end_date;
-      }
-    } else if (this.props.dataSource.school) {
+    return companyName;
+  }
+
+  getPosition() {
+    let position = this.state.position;
+
+    if (this.props.dataSource.position) {
+      position = this.props.dataSource.position.name;
+      return position;
+    }
+
+    return position;
+  }
+
+  getPeriod() {
+    let period = this.state.period;
+
+    if (this.props.dataSource.start_date) {
+      period = this.props.dataSource.start_date + ' - ' + this.props.dataSource.end_date;
+      return period;
+    }
+
+    return period;
+  }
+
+  getSchoolName() {
+    let schoolName = this.state.schoolName;
+
+    if (this.props.dataSource.school) {
       schoolName = this.props.dataSource.school.name;
-      identifier = 2;
-      if (this.props.dataSource.concentration.length > 0) {
-        major = this.props.dataSource.concentration[0].name;
-      }
+      return schoolName;
+    }
 
-      if (this.props.dataSource.year) {
-        period = 'Class of ' + this.props.dataSource.year.name;
-      }
+    return schoolName;
+  }
+
+  getMajor() {
+    let major = this.state.major;
+
+    if (this.props.dataSource.concentration.length > 0) {
+      major = this.props.dataSource.concentration[0].name;
+      return major;
     }
 
-    this.setState({
-      schoolName: schoolName,
-      major: major,
-      companyName: companyName,
-      position: position,
-      period: period,
-      identifier: identifier,
-      loaded: true,
-    });
+    return major;
   }
 
-  renderEducation() {
+  graduationYear() {
```

Comment:
How about rename this function to have a verb form? Traditionally we use verb when make a function name